compiler/luci-codegen/README.md

```diff
@@ -0,0 +1,203 @@
+# luci-codegen
+
+## Overview
+
+`luci-codegen` component is responcible for native code generation from graph representation.
+
+Package provides:
+- Library `luci_codegen`: takes luci IR, replaces some nodes with special "Compiled" custom nodes.
+- Executable `circle_codegen`: takes neural network in circle format, generates new circle model with compiled subgraphs. 
+
+## Build
+
+This component needs external `Halide` build. If `Halide` is built from sources it demands `LLVM`.
+
+To get needed components and build `luci_codegen` library and `circle_codegen` executable on linux you can use following instruction.
+Also see README.md in Halide repo for more methods to build LLVM and Halide (including windows build).
+
+##### LLVM Build
+
+Tested on commit **24faa87075ac1b4e4d7f7336271b1ba72ffcc2c0** from Jan 12 (master branch)
+
+```
+$ git clone https://github.com/llvm/llvm-project.git
+$ cd llvm-project
+# mkdir build; cd build
+$ cmake -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=../install \
+        -DLLVM_ENABLE_PROJECTS="clang;lld;clang-tools-extra" \
```

Comment:
Do we really need `clang-tools-extra`?

Reply:
I think I tried to build without this target, but for some reason Halide did not accept this llvm.
Though this was some time ago, so I'll try again